cloud/resource_apikey.go

```diff
@@ -0,0 +1,353 @@
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["apikey_name"],
+				ValidateFunc: validateNotBlank,
+			},
+			"instance_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions["instance_name"],
+			},
+			"service_account_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions["service_account_name"],
+			},
+			"expiration_time": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions["expiration_time"],
+			},
+			"revoke": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: descriptions["revoke"],
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: descriptions["description"],
+				Optional:    true,
+			},
+			"ready": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["apikey_ready"],
+			},
+			"issued_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["issued_at"],
+			},
+			"expires_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["expires_at"],
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["private_key"],
+			},
+			"key_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["key_id"],
+			},
+			"revoked_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions["revoked_at"],
+			},
+		},
+	}
+}
+
+func resourceApiKeyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	namespace := d.Get("organization").(string)
+	name := d.Get("name").(string)
+	instanceName := d.Get("instance_name").(string)
+	serviceAccountName := d.Get("service_account_name").(string)
+	description := d.Get("description").(string)
+	expirationTime := d.Get("expiration_time").(string)
+	clientSet, err := getClientSet(getFactoryFromMeta(m))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("ERROR_INIT_CLIENT_ON_CREATE_API_KEY: %w", err))
+	}
+	ak := &v1alpha1.APIKey{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "APIKey",
+			APIVersion: v1alpha1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.APIKeySpec{
+			InstanceName:       instanceName,
+			ServiceAccountName: serviceAccountName,
+		},
+	}
+	r1 := regexp.MustCompile(`^(\d+.)(s|m|h|d)$`)
+	t := time.Now()
+	if expirationTime != "" {
+		if r1.MatchString(expirationTime) {
+			ago, err := str2duration.ParseDuration(expirationTime)
+			if err != nil {
+				return diag.FromErr(fmt.Errorf("ERROR_PARSE_EXPIRATION_TIME: %w", err))
+			}
+			t = t.Add(ago)
+		} else if expirationTime != "No expiration date" {
+			layout := "02/01/2006 15:04:05"
```

Comment:
We need to make them consistent with `console` and `snctl` by using the format `2006-01-02T15:04:05Z`

Reply:
In the console it seems to use this format @urfreespace 
![image](https://github.com/streamnative/terraform-provider-streamnative/assets/1907867/0ae8afbf-551c-44db-9daa-915a7a6d5c3f)

Reply:
Fixed, perhaps we should also update the console to use this time format `2006-01-02T15:04:05Z`